### DISCUSSION TITLE
Notes Jan 11th, 2023

### DISCUSSION THREAD

**Discussion by wipfli** [maplibre/maplibre#187] [Technical Steering Committee Meetings]
# January 11th, 2023: Native

7 PM to 8 PM CET

### Attendees (First Name, Last Name, GitHub handle, affiliation)

* Bart, Louwers, @louwers, Maintainer Native MapLibre
* Steve Gifford @sjg-wdw - Wet Dog Weather
* Tadej Novak, @ntadej, Independent
* Birk Skyum, @birkskyum, MapLibre Board Member,
* Luke Seelenbinder, @lseelenbinder, Stadia Maps / MapLibre Board Member
* Max Ammann, independent
* Swagata Prateek, AWS
* Damon Burgett, Stamen
* Haowen You, AWS
* Jesse Levine, Microsoft
* Marc Wilson, Meta
* Marc Regan, Meta
* Nathan Van Fleet, Lyft
* Ovidiu Voda, MapLibre Native Maintainer
* Stephanie May, Stamen
* Steve Gifford, Wet Dog Weather
* Tristram Gräbener, @tristramg Codeurs en Liberté/cocarto.com
* Yuri Astrakhan, MapLibre Governing Board

Chair: Luke Seelenbinder

### MapLibre GL Native Organizational Questions

* [wipfli] Bart Louwers started as a maintainer for native with us, see [https://maplibre.org/news/2023-01-03-announcement-bart/](https://maplibre.org/news/2023-01-03-announcement-bart/). Next week, Ovidiu Voda(Ovi) will also join in the maintainer native role and support Bart (announcement will follow). Please welcome both!
* [sjg] Metal Port Design Proposal PR 
    * 2 Technical Reviews - Done
    * 2 Board Reviews - Done
    * Ready to merge
* [ntadej] Volunteers to write the repository split design proposal, basic idea:
    * Have a maplibre-core (or maplibre-renderer) base repository which can do headless rendering on all platforms. Runs exactly the same tests for all platforms. Fully C++ based (maybe a bit of objective-C for Apple platforms). Maintains feature-parity among platforms.
    * SDKs for individual platforms go in separate repositories. They implement platform-specific bindings and other functionality currently available. Functionality may differ between platforms. \
Feedback:
        * Bart: 
            * this can increase the complexity
            * we should state the goals
        * Prateek: coming from android dev, splitting into multiple repos, looks like a good solution especially for CI/CD part
        * Luke: Define what is core.

        Conclusion: We will move the discussion to GitHub

* [ntadej] There are many branches in the repo. The policy is to do all the development in forks. Should we enforce it more (it makes it clearer)?
* [lseelenbinder] [https://github.com/maplibre/maplibre-react-native](https://github.com/maplibre/maplibre-react-native) is ready for use.
* [prateek] issue cleanup
    * What do we do with old PRs?

### MapLibre Organizational Questions

* [bart] [What projects part of the maplibre-gl-native repository are core projects?](https://github.com/maplibre/maplibre/discussions/186)
    * macOS could be part of the common iOS/Apple SDK, but will not be core the project

# January 11th, 2023: Web

8 PM to 9 PM CET

### Attendees (First Name, Last Name, GitHub handle, affiliation)

* Harel Mazor @HarelM Israel Hiking Map/Maplibre board member
* Luke Seelenbinder @lseelenbinder - Stadia Maps / MapLibre board member
* Tristram Gräbener, @tristramg Codeurs en Liberté/cocarto.com
* Derek Westcott @drwestco Microsoft / Bing / Azure Maps
* Oliver Wipfli @wipfli MapLibre Coordinator
* Bart Louwers, MapLibre native maintainer
* Birk Skyum, MapLibre Governing Board member
* Max Ammann, independent
* Ovi Voda, MapLibre Native maintainer
* Xiaoji Chen, Microsoft Research and deck.gl and react-map-gl maintainer
* Yuri Astrakhan, MapLibre Governing Board member

Chair: Harel Mazor

### MapLibre GL JS Technical Questions

* [HarelM] [https://github.com/maplibre/maplibre-gl-js/pull/1800](https://github.com/maplibre/maplibre-gl-js/pull/1800) - Add contentMatch to sprites - waiting for poll results: [https://github.com/maplibre/maplibre-gl-js/discussions/1944](https://github.com/maplibre/maplibre-gl-js/discussions/1944)
    * Waiting for Microsoft to update the PR
* [HarelM] [https://github.com/maplibre/maplibre-gl-js/pull/2038](https://github.com/maplibre/maplibre-gl-js/pull/2038) - moving to promises - heads up
    * Agreed this is a good direction
* [HarelM] [https://github.com/maplibre/maplibre-gl-js/discussions/1851](https://github.com/maplibre/maplibre-gl-js/discussions/1851) - style spec language and validation - continue discussion from last meeting?
    * JSON schema might be a better directions
    * There’s a need to check what part the Native is using (@Bart/@Ovi)
    * Docs site generation might be impacted and is currently a problem now as the tools we are using are outdated and some are open source
* [wipfli] Breaking changes needed in v3 for reactive frameworks?
    * [https://github.com/maplibre/maplibre-gl-js/issues/1545](https://github.com/maplibre/maplibre-gl-js/issues/1545)
    * Harel: Make design proposal discussion before pull request
    * Harel: Map inherits from Camera. Maybe we should change that to Map having a Camera object.
    * Xiaoji: deck.gl plans to switch to maplibre
    * Harel: v3 comes probably in 2 months, Xiaoji: can work on it 
* [wipfli] What is the webgl2 status?
    * Xiaoji: deck.gl uses headless chrome for webgl2 tests
    * Harel: We tried this too, but seems too slow, in particular for local development.
    * Xiaoji: Provide good way for running restricted set of tests.
* [xiaoji]: Do you consider switching to pure es module?
    * Birk: May be possible for v3

The next monthly meeting is on Wednesday, Feb 8th, 2023 (as always on the second Wednesday of the month). Feel free to join next time, and feel free to add any comments or questions below...

**Comment by Ovi** [comment]
Just making sure you mentioned the " @Ovi " you were supposed to? :smile:

**Reply by wipfli** [reply]
Good point, thanks @Ovi, I guess Harel was not aware that I would post these notes on GitHub... But feel free to join our community, @Ovi!
